backend/src/api/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using api.Data.DTO;
+using api.Services.Users;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+
+namespace api.Controllers;
+
+[ApiController]
+[Route("/api/auth")]
+public class AuthenticationController : ControllerBase
+{
+    private readonly IUserService _userService;
+
+    public AuthenticationController(IUserService userService)
+    {
+        _userService = userService;
+    }
+
+    [HttpPost("register")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(400)]
+    public async Task<IActionResult> Register([FromBody] UserDTO user)
+    {
+        if (!ModelState.IsValid) { return BadRequest("Invalid data provided!"); }
+
+        var (IsSuccess, Error) = await _userService.Register(user);
+        if (IsSuccess)
+        {
+            return Ok("User created");
+        }
+        else
+        {
+            return BadRequest(Error);
+        }
+    }
+
+    [HttpPost("login")]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(401)]
```

Comment:
Shouldn't there be also `[ProducesResponseType(400)]` as there is a possibility of `BadRequest` (no login credentials)?

Reply:
It is only used to creat a Swagger document not necessary, good to add